connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -195,10 +209,91 @@ public Connector connector() {
                 "id=" + connName +
                 '}';
     }
+
+    class ConnectorMetricsGroup {
+        private final MetricGroup metricGroup;
+        private final MetricName statusRunningName;
+        private final MetricName statusPausedName;
+        private final MetricName statusFailedName;
+        private final Sensor statusRunning;
+        private final Sensor statusPaused;
+        private final Sensor statusFailed;
+
+        public ConnectorMetricsGroup(ConnectMetrics connectMetrics) {
+            this.metricGroup = connectMetrics.group("connector-metrics",
+                    "connector", connName);
+
+            this.statusRunningName = metricGroup.metricName("status-running",
+                    "Signals whether the connector is in the running state.");
+            this.statusRunning = metricGroup.metrics().sensor("status-running");
+            this.statusRunning.add(statusRunningName, new Value());
+
+            this.statusPausedName = metricGroup.metricName("status-paused",
+                    "Signals whether the connector is in the paused state.");
+            this.statusPaused = metricGroup.metrics().sensor("status-paused");
+            this.statusPaused.add(statusPausedName, new Value());
+
+            this.statusFailedName = metricGroup.metricName("status-failed",
+                    "Signals whether the connector is in the failed state.");
+            this.statusFailed = metricGroup.metrics().sensor("status-failed");
+            this.statusFailed.add(statusFailedName, new Value());
+
+            recordInitialized();
+        }
+
+        public void recordInitialized() {
+            this.statusRunning.record(0);
+            this.statusPaused.record(0);
+            this.statusFailed.record(0);
+        }
+
+        public void recordStopped() {
+            this.statusRunning.record(0);
+            this.statusPaused.record(0);
+            this.statusFailed.record(0);
+        }
+
+        public void recordRunning() {
+            this.statusRunning.record(1);
+            this.statusPaused.record(0);
+            this.statusFailed.record(0);
+        }
+
+        public void recordPaused() {
+            this.statusRunning.record(0);
+            this.statusPaused.record(1);
+            this.statusFailed.record(0);
+        }
+        public void recordFailed() {
+            this.statusRunning.record(0);
+            this.statusPaused.record(0);
+            this.statusFailed.record(1);
+        }
+        public boolean isRunning() {
+            return asBoolean(statusRunningName);
+        }
+
+        public boolean isPaused() {
+            return asBoolean(statusPausedName);
+        }
+
+        public boolean isFailed() {
+            return asBoolean(statusFailedName);
+        }
+
+        private boolean asBoolean(MetricName metric) {
+            double value = this.metricGroup.metrics().metric(metric).value();
+            return value > 0.00001d || value < -0.000001d;
```

Comment:
Radius doesn't seem to be equal on the positive and negative side. 
Should this be: 
```java
private static final double EPSILON = 0.00001d;
...
return value > EPSILON || value < -EPSILON;
```
instead? (also not sure which accuracy you prefer for epsilon `0.00001d` or `0.000001d`?)

Reply:
Great catch. +1